src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx

```diff
@@ -0,0 +1,1173 @@
+   0.26945,0.27820,0.28870,0.29741,0.30783,0.31656,0.32704,0.33571,
+   0.34647,0.35516,0.36611,0.37488,0.38589,0.39511,0.40656,0.41554,
+   0.42744,0.43665,0.44867,0.45803,0.47063,0.47997,0.49291,0.50236,
+   0.51581,0.52528,0.53889,0.54866,0.56283,0.57241,0.58686,0.59671,
+   0.61154,0.62125,0.63630,0.64631,0.66182,0.67149,0.68740,0.69703,
+   0.71343,0.72264,0.73945,0.74866,0.76553,0.77444,0.79212,0.80021,
+   0.81841,0.82593,0.84434,0.85145,0.87018,0.87644,0.89563,0.90090,
+   0.92047,0.92494,0.94450,0.94782,0.96768,0.96975,0.98974,1.0};
+  
+  double W2=W*W;
+  double MN = fAM;
+  double MN2 = MN*MN;
+  double MD = fMD;
+  double Mp = fMP;
+  double Mp2 = Mp*Mp;
+  double nu = (W2 - MN2 + Q2)/2./MN;
+  double qv = TMath::Sqrt(nu*nu + Q2);
+  F1 = 0.;
+  R = 0.;
+  sigmaT = 0.;
+  sigmaL = 0.;
+  // Do fast 20 bins if abvoe threshold
+  if(W2>1.30)
+  {
+    for (int ism = 0; ism<20; ism++)
+    {
+      double W2p = TMath::Power(MD + nu - sqrt(MN2 + avp2[ism]),2) - qv*qv + 2.*qv*avpz[ism] - avp2[ism];
+      if(W2p>1.155)
+      {
+         double Wp = TMath::Sqrt(W2p);
+         double sigtp = sigmaR(0, Q2, Wp, isDeuterium) + sigmaNR(0, Q2, Wp, isDeuterium);
+         double F1p = sigtp*(W2p-Mp2)/8./kPi2/kAem;
+         F1 += F1p*fyd[ism]/10.;
+         if (!isDeuterium)
+         {
+            double siglp = sigmaR(1, Q2, Wp) + sigmaNR(1, Q2, Wp); 
+            sigmaL += siglp*fyd[ism]/10.;
+            sigmaT += sigtp*fyd[ism]/10.;
+         }
+      }
+    }
+  }
+  else
+  {
+     for (int ism = 0;ism<200;ism++)
+     {
+       double pz = -1. + 0.01*(ism + 0.5);
+       // Need avp2f term to get right behavior x>1! 
+       double W2p = TMath::Power(MD + nu - sqrt(MN2 + avp2f[ism]),2) - qv*qv + 2.*qv*pz - avp2f[ism];
+       if(W2p>1.155)
+       {
+          double Wp = TMath::Sqrt(W2p);
+          double sigtp = sigmaR(0, Q2, Wp, isDeuterium) + sigmaNR(0, Q2, Wp, isDeuterium);
+          double F1p = sigtp*(W2p-Mp2)/8./kPi2/kAem;
+          F1 += F1p*fydf[ism]/100.;
+          if (!isDeuterium)
+          {
+            double siglp = sigmaR(1, Q2, Wp) + sigmaNR(1, Q2, Wp); 
+            sigmaT += sigtp*fydf[ism]/100.;
+            sigmaL += siglp*fydf[ism]/100.;
+          }
+       }
+     }
+  }
+  if (isDeuterium && fUseMEC)
+       F1 += fMECcoef[0]*TMath::Exp(-(W - fMECcoef[1])*(W - fMECcoef[1])/fMECcoef[2])/
+             TMath::Power(1. + TMath::Max(0.3,Q2)/fMECcoef[3],fMECcoef[4])*TMath::Power(nu, fMECcoef[5]);
+  if(!isDeuterium && sigmaT!=0.) 
+    R = sigmaL/sigmaT;
+  
+}
+//____________________________________________________________________________
+double BostedChristyEMPXSec::MEC2009(int A, double Q2, double W) const
+{
+  double F1 = 0.0;
+  double W2 = W*W;
+  double Mp = fAM;
+  double Mp2 = Mp*Mp;
+  if(W2<=0.0)
+    return F1;
+  double nu = (W2 - Mp2 + Q2)/2./Mp;
+  double x  = Q2/2.0/Mp/nu;
+  
+  if(A<=2) 
+    return F1;
+  
+  double p18 = 206.13;
```

Comment:
Also here, can we add comments about all these hardcoded values?

Reply:
Move to config file.